designer/server/src/main/scala/pl/touk/nussknacker/ui/api/description/scenarioActivity/Dtos.scala

```diff
@@ -0,0 +1,528 @@
+    }
+
+    case object IncomingMigration extends ScenarioActivityType {
+      override def displayableName: String        = "Incoming migration"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List("compare")
+    }
+
+    case object OutgoingMigration extends ScenarioActivityType {
+      override def displayableName: String        = "Outgoing migration"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object PerformedSingleExecution extends ScenarioActivityType {
+      override def displayableName: String        = "Processing data"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object PerformedScheduledExecution extends ScenarioActivityType {
+      override def displayableName: String        = "Processing data"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    case object AutomaticUpdate extends ScenarioActivityType {
+      override def displayableName: String        = "Automatic update"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List("compare")
+    }
+
+    case object CustomAction extends ScenarioActivityType {
+      override def displayableName: String        = "Custom action"
+      override def icon: String                   = "/assets/states/error.svg"
+      override def supportedActions: List[String] = List.empty
+    }
+
+    override def values: immutable.IndexedSeq[ScenarioActivityType] = findValues
+
+    implicit def scenarioActivityTypeSchema: Schema[ScenarioActivityType] =
+      enumSchema[ScenarioActivityType](
+        ScenarioActivityType.values.toList,
+        _.entryName,
+      )
+
+    implicit def scenarioActivityTypeCodec: circe.Codec[ScenarioActivityType] = circe.Codec.from(
+      Decoder.decodeString.emap(str =>
+        ScenarioActivityType.withNameEither(str).left.map(_ => s"Invalid scenario action type [$str]")
+      ),
+      Encoder.encodeString.contramap(_.entryName),
+    )
+
+    implicit def scenarioActivityTypeTextCodec: Codec[String, ScenarioActivityType, CodecFormat.TextPlain] =
+      Codec.string.map(
+        Mapping.fromDecode[String, ScenarioActivityType] {
+          ScenarioActivityType.withNameOption(_) match {
+            case Some(value) => DecodeResult.Value(value)
+            case None        => DecodeResult.InvalidValue(Nil)
+          }
+        }(_.entryName)
+      )
+
+  }
+
+  @derive(encoder, decoder, schema)
+  final case class ScenarioActivityComment(comment: Option[String], lastModifiedBy: String, lastModifiedAt: Instant)
```

Comment:
Because it may be deleted, and is then unavailable. But it wasn't the best representation, so I changed the dto structure to represent this situation:

- instead of nullable field `"comment": "Added new processing step"`
- there will be 
```
"status": {
  "type": "AVAILABLE",
  "comment": "Added new processing step"
}

or 

"status": {
  "type": "DELETED"
}
```